src/entity/posts/utils.ts

```diff
@@ -340,6 +341,56 @@ export const createSharePost = async (
   }
 };
 
+export const updateSharePost = async (
+  con: DataSource | EntityManager,
+  userId: string,
+  postId: string,
+  sourceId: string,
+  commentary: string | null,
+) => {
+  let strippedCommentary = commentary;
+
+  if (commentary?.length) {
+    strippedCommentary = commentary.trim();
+    await validateCommentary(strippedCommentary);
+  } else {
+    strippedCommentary = null;
+  }
+
+  try {
+    const mentions = await getMentions(con, commentary, userId, sourceId);
+    const titleHtml = commentary?.length
+      ? generateTitleHtml(commentary, mentions)
+      : null;
+
+    await con.getRepository(SharePost).update(
+      {
+        id: postId,
+        authorId: userId,
+      },
+      {
+        title: strippedCommentary,
+        titleHtml,
+      },
+    );
+
+    if (mentions.length) {
+      await saveMentions(con, postId, userId, mentions, PostMention);
+    }
```

Comment:
What happens when we remove mentions on post edit, we should remove them right?

Reply:
that's a good point, and as far as I can see, this use case is not being covered on the `editPost` mutation either -- maybe we should fix this there as well?

Reply:
after [discussing in slack](https://dailydotdev.slack.com/archives/C02E2C3C13R/p1697103348449099), it seems not removing mentions is intentional, for now